src/orchestration/orchestration.cpp

```diff
@@ -0,0 +1,911 @@
+    ERR_FAIL_COND_V_MSG(_variables.has(p_method.name), nullptr, "A variable already exists with the name: " + p_method.name);
+    ERR_FAIL_COND_V_MSG(_signals.has(p_method.name), nullptr, "A signal already exists with the name: " + p_method.name);
+
+    Ref<OScriptFunction> function(memnew(OScriptFunction));
+    function->_orchestration = this;
+    function->_guid = Guid::create_guid();
+    function->_method = p_method;
+    function->_owning_node_id = p_node_id;
+    function->_user_defined = p_user_defined;
+
+    _functions[p_method.name] = function;
+
+    _self->emit_signal("functions_changed");
+
+    return function;
+}
+
+void Orchestration::remove_function(const StringName& p_name)
+{
+    ERR_FAIL_COND_MSG(_has_instances(), "Cannot remove functions, instances exist.");
+    ERR_FAIL_COND_MSG(!_functions.has(p_name), "Cannot remove function that does not exist with name: " + p_name);
+    {
+        Ref<OScriptFunction> function = _functions[p_name];
+        ERR_FAIL_COND_MSG(!function.is_valid(), "Function found with name '" + p_name + "', but its not valid.");
+
+        // Check if the function has a graph (user-defined functions do)
+        if (_graphs.has(p_name))
+        {
+            Ref<OScriptGraph> graph = _graphs[p_name];
+            if (graph->get_flags().has_flag(OScriptGraph::GF_FUNCTION))
+                remove_graph(graph->get_graph_name());
+        }
+
+        const List<int> node_ids = _get_node_type_node_ids<OScriptNodeCallScriptFunction>();
+        for (int node_id : node_ids)
+        {
+            const Ref<OScriptNodeCallScriptFunction> call_function = get_node(node_id);
+            const Ref<OScriptFunction> called_function = call_function->get_function();
+            if (call_function.is_valid() && called_function->get_function_name().match(function->get_function_name()))
+                remove_node(node_id);
+        }
+
+        // Find the node for this function and remove it
+        if (has_node(function->get_owning_node_id()))
+            remove_node(function->get_owning_node_id());
+
+        // Let the editor handle node removal
+        _functions.erase(p_name);
+
+        _self->emit_signal("functions_changed");
+    }
+    _self->emit_changed();
+}
+
+Ref<OScriptFunction> Orchestration::find_function(const StringName& p_name) const
+{
+    for (const KeyValue<StringName, Ref<OScriptFunction>>& E : _functions)
+    {
+        if (E.value->get_function_name().match(p_name))
+            return E.value;
+    }
+    return nullptr;
+}
+
+Ref<OScriptFunction> Orchestration::find_function(const Guid& p_guid) const
+{
+    for (const KeyValue<StringName, Ref<OScriptFunction>>& E : _functions)
+    {
+        if (E.value->get_guid() == p_guid)
+            return E.value;
+    }
+    return nullptr;
+}
+
+void Orchestration::rename_function(const StringName& p_old_name, const StringName& p_new_name)
```

Comment:
I wonder if `rename_xxxx` should be done through the XXXX target object and friend delegate to `Orchestration`.  What about other resource types.